src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -92,6 +94,8 @@
                 MultiBinding.MultiEnvironment environment = binding.bind(run, workspace, launcher, listener);
                 unbinders.add(environment.getUnbinder());
                 overrides.putAll(environment.getValues());
+                IdCredentials id = CredentialsProvider.findCredentialById(binding.getCredentialsId(), binding.type(), run);
+                CredentialsProvider.track(run, id);
```

Comment:
Well, not in the case of Pipeline builds generally, but I see your point—`track(Node, Credentials)` is irrelevant here. I will file a PR to fix the Javadoc.